ducktape/cluster/remoteaccount.py

```diff
@@ -142,97 +317,182 @@ def kill_process(self, process_grep_str, clean_shutdown=True, allow_fail=False):
 
-        r += src + " " + dest
-        return r
+        Example:
 
-    def scp_from(self, src, dest, recursive=False):
-        """Copy something from this node. src may be a string or an iterable of several sources."""
-        return self._ssh_quiet(self.scp_from_command(src, dest, recursive))
+        path/to/file, path/to/renamed_file, node
+            file will be copied to renamed_file on node
 
-    def scp_to_command(self, src, dest, recursive=False):
-        if not isinstance(src, basestring):
-            # Assume src is iterable
-            src = " ".join(src)
+        """
+        # TODO: if dest is an existing file, what is the behavior?
 
-        r = "scp "
-        if self.ssh_args:
-            r += self.ssh_args + " "
-        if recursive:
-            r += "-r "
-        r += src + " "
-        if self.user:
-            r += self.user + "@"
-        r += self.hostname + ":" + dest
-        return r
+        temp_dir = tempfile.mkdtemp()
 
-    def scp_to(self, src, dest, recursive=False):
-        return self._ssh_quiet(self.scp_to_command(src, dest, recursive))
+        try:
+            src_name = src
+            if src_name.endswith(os.path.sep):
+                src_name = src_name[:-len(os.path.sep)]  # trim off path separator from end
+            src_name = os.path.basename(src_name)
+
+            # TODO: deal with very unlikely case that src_name matches temp_dir name?
+            # TODO: I think this actually works
+
+            local_dest = os.path.join(temp_dir, src_name)
+            self.copy_from(src, local_dest)
+
+            dest_node.account.copy_to(local_dest, dest)
+
+        finally:
+            if os.path.isdir(temp_dir):
+                shutil.rmtree(temp_dir)
+
+    def copy_from(self, src, dest):
+        if os.path.isdir(dest):
+            # dest is an existing directory, so assuming src looks like path/to/src_name,
+            # in this case we'll copy as:
+            #   path/to/src_name -> dest/src_name
+            src_name = src
+            if src_name.endswith(os.path.sep):
+                src_name = src_name[:-len(os.path.sep)]  # trim off path separator from end
+            src_name = os.path.basename(src_name)
+
+            dest = os.path.join(dest, src_name)
+
+        if self.isfile(src):
+            self.sftp_client.get(src, dest)
+        elif self.isdir(src):
+            # we can now assume dest path looks like: path_that_exists/new_directory
+            os.mkdir(dest)
+
+            # for obj in `ls src`, if it's a file, copy with copy_file_from, elif its a directory, call again
+            for obj in self.sftp_client.listdir(src):
+                obj_path = os.path.join(src, obj)
+                if self.isfile(obj_path) or self.isdir(obj_path):
+                    self.copy_from(obj_path, dest)
+                else:
+                    # TODO what about uncopyable file types?
+                    pass
+
+    def copy_to(self, src, dest):
```

Comment:
Then it becomes effectively a static method, and I think it would be irritating to import `RemoteAccount` to use it.

If it's a method, you're effectively pinning `src_host` or `dest_host` to `self`, but then you need a modifier e.g. `to`/`from`/`between` to clarify which is which.